portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,33 +30,45 @@
+/** Servlet that adds a comment to the datastore. */
 @WebServlet("/contact-me")
 public class DataServlet extends HttpServlet {
 
-    @Override
-    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        // Get the input from the form.
-        String name = request.getParameter("name-input");
-        String email = request.getParameter("email-input");
-        String subject = request.getParameter("subject-input");
-        String message = request.getParameter("message-input");
-        long timestamp = System.currentTimeMillis();
-
-        // Error handling - don't allow empty or null values
-        if (!Strings.isNullOrEmpty(name) && !Strings.isNullOrEmpty(email) && !Strings.isNullOrEmpty(subject) && !Strings.isNullOrEmpty(message)) {
-            Entity commentEntity = new Entity("Comment");
-            commentEntity.setProperty("name", name);
-            commentEntity.setProperty("email", email);
-            commentEntity.setProperty("subject", subject);
-            commentEntity.setProperty("message", message);
-            commentEntity.setProperty("timestamp", timestamp);
-
-            DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-            datastore.put(commentEntity);
-        }
-
-        // Redirect back to the HTML page.
-        response.sendRedirect("/index.html");
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String name = request.getParameter("name-input");
+    String email = request.getParameter("email-input");
+    String subject = request.getParameter("subject-input");
+    String message = request.getParameter("message-input");
+    long timestamp = System.currentTimeMillis();
+
+    LanguageServiceClient languageService = LanguageServiceClient.create();
+    double sentimentScore = getSentimentScore(message, languageService);
+    languageService.close();
+
+    // Error handling - don't allow empty or null values
+    if (!Strings.isNullOrEmpty(name) && !Strings.isNullOrEmpty(email) && !Strings.isNullOrEmpty(subject) && !Strings.isNullOrEmpty(message)) {
+      Entity commentEntity = new Entity("Comment");
+      commentEntity.setProperty("name", name);
+      commentEntity.setProperty("email", email);
+      commentEntity.setProperty("subject", subject);
+      commentEntity.setProperty("message", message);
+      commentEntity.setProperty("timestamp", timestamp);
+      commentEntity.setProperty("sentimentScore", sentimentScore);
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(commentEntity);
     }
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
+  }
+
+  double getSentimentScore(String message, LanguageServiceClient languageService) throws IOException {
```

Comment:
We try to minimize the accessibility of classes and members so that we can separate the implementation details from the API. So since this is not part of the API and an implementation detail, I would make this method private.